miner/miner.go

```diff
@@ -60,31 +100,69 @@ func Mine(bc *blockchain.BlockChain, b *blockchain.Block) *MiningResult {
 		b.Nonce++
 	}
 
+	setState(Stopped)
 	return &MiningResult{
 		Complete: true,
 		Info:     MiningSuccessful,
 	}
 }
 
-func setStart() {
-	currentlyMiningLock.Lock()
-	defer currentlyMiningLock.Unlock()
-	currentlyMining = true
+// StopMining causes the miner to abort the current mining job immediately.
+func StopMining() {
```

Comment:
I would like to see these incorporated into a struct like we did with app. Package level state is an antipattern, even if you only expect that state to be instantiated once